server/store/jdbc/src/main/java/com/alipay/sofa/registry/jdbc/exception/RevisionNotExistException.java

```diff
@@ -20,7 +20,7 @@
  * @author xiaojian.xj
  * @version $Id: RevisionNotExistException.java, v 0.1 2021年02月03日 15:59 xiaojian.xj Exp $
  */
-public class RevisionNotExistException extends RuntimeException {
+public class RevisionNotExistException extends Error {
```

Comment:
Why is this exception actually an error?

Reply:
The original RevisionNotExistException inherits from RuntimeException and is not caught, causing errors in logging that interfere with troubleshooting. Here, it is modified to Error and actively handled.

Reply:
[doge] 
1. if `RevisionNotExistException` extends from `Error`, it better be a `RevisionNotExistError` 
2. if `RevisionNotExistException` is not caught, `Error` is much more dangerous, as an `Error` could only be caught by `catch(Throwable)` not `catch(Exception)`
[doge]

Reply:
ok, RevisionNotExistException catch ( ExecutionError ) when cache.get().  we renamed it to "RevisionNotExistError".

Reply:
The com.google.common.cache.LoadingCache#get clearly states the following, I think the reason RevisionNotExistException was not caught is because we didn't catch the UncheckedExecutionException.

> Throws:
ExecutionException – if a checked exception was thrown while loading the value. (ExecutionException is thrown even if computation was interrupted by an InterruptedException.)
UncheckedExecutionException – if an unchecked exception was thrown while loading the value
ExecutionError – if an error was thrown while loading the value

Reply:
By the way, it appears that RevisionNotExist more closely resembles an unchecked exception rather than a checked exception or an error.

Reply:
how about just change  from catch(ExecutionException) to change(Throwable), so that RevisionNotExistException can be catch.

Reply:
better to handle both `ExecutionException` and `Throwable`, one for exclusive error handling, and the other for general error handling

Reply:
Either option is acceptable, provided that we manage the RevisionNotExistException to deliver more precise error messages.

Reply:
follow Nocvalight's suggestion.